src/use-swr.ts

```diff
@@ -262,7 +261,8 @@ export const useSWRHandler = <Data = any, Error = any>(
           broadcastState(cache, key, newData, newState.error, false)
         }
       } catch (err) {
-        cleanupState()
+        // Reset the state immediately.
+        cleanupState(startAt!)
```

Comment:
Maybe give startAt a default value -1 or disable the lint rule `Forbidden non-null assertion`

Reply:
Sure!

Reply:
I decided to disable that line only since the rule is overall helpful.